tests/admin_widgets/tests.py

```diff
@@ -1142,6 +1142,44 @@ def test_filter(self):
         self.assertEqual(list(self.school.alumni.all()),
                          [self.jason, self.peter])
 
+    def test_refresh_page(self):
+        """
+        Test case for #22955
+        Needs to be sure that horizontal and vertical filters keep selected
+        options when user deselect them and refreshes the page.
+        """
+        self.admin_login(username='super', password='secret', login_url='/')
+        self.selenium.get(
+            '%s%s' % (self.live_server_url, reverse('admin:admin_widgets_school_change', args=(self.school.id,))))
+
+        for field_name in ['students', 'alumni']:
+            from_box = '#id_%s_from' % field_name
+            choose_link = '#id_%s_add_link' % field_name
+
+            self.get_select_option(from_box, str(self.arthur.id)).click()
+            self.get_select_option(from_box, str(self.jason.id)).click()
+            self.selenium.find_element_by_css_selector(choose_link).click()
+
+        self.selenium.find_element_by_xpath('//input[@name="_continue"]').click()
+        self.wait_page_loaded()
+
+        to_box = '#id_alumni_to'
+        remove_link = '#id_alumni_remove_all_link'
+
+        options_len = len(self.selenium.find_elements_by_css_selector(to_box + ' > option'))
+        self.assertEqual(options_len, 2)
+        self.selenium.find_element_by_css_selector(remove_link).click()
```

Comment:
This doesn't seem to be necessary to reproduce the bug. Just refreshing the page without making any changes caused the select options to clear.